### PR TITLE
Added npm.entry

### PR DESCRIPTION
**Pull request by taras**
The code that I want to test is using `require.resolve`, I'd like to test it using fixtures but `require.resolve` doesn't find my fixtures. I tried 2 mocking libraries [require-mocked](https://www.npmjs.com/package/require-mocked) and [rewire](https://www.npmjs.com/package/rewire) but they both seem to interfere with require's functionality in their own way. 

How can I mock `require.resolve`?

### CODE REVIEW

File: lib/npm.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+var find = require('./find');
+var rename = require('./rename');
+
+module.exports = {
+  entry: entry,
+  entryPath: entryPath
+};
+
+/**
+ * Return tree containing the file referenced in package.json.
```
Comment (stefanpenner) [inline lib/npm.js]: can you add an example

File: lib/npm.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  if (newName) {
+    return rename(foundTree, function(){
+      return newName;
+    })
+  }
+  return foundTree;
+}
+
+/**
+ * Return path to script referenced in package.json#main
```
Comment (stefanpenner) [inline lib/npm.js]: can you add an example

Comment (taras) [inline lib/npm.js]: done

File: lib/npm.js (hunk 1 above)
Comment (taras) [inline lib/npm.js]: done

File: lib/npm.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ * }
+ *
+ * entryPath('foo') -> '/absolute/path/to/module/index.js'
+ *
+ * @param moduleName
+ * @returns {string}
+ */
+function entryPath(moduleName) {
+  var modulePath = resolve(moduleName);
+  var pkg = require(p.join(modulePath, 'package.json'));
```
Comment (stefanpenner) [inline lib/npm.js]: the entry file is not always going to be at the root.

**Comment by stefanpenner** [comment]
we clearly need something handy like this. Its quite common to want to just pluck the main. At some point @chadhietala work will provide  us a more advance solution.

quick thought to an implementation:

```js
function npmMain(package, name) {
  var resolved = resolveMain(package);

  return stew.rename(stew.find(resolved), function() {
    return name || package;
  });
}

function resolveMain(package) {
  var fullPath = require.resolve(package);
  var path = require('path');

  return path.dirname(fullPath) + '{' + path.basename(fullPath) + '}';
}
```

```js

merge([
  npmMain('loader.js')
  npmMain('loader.js', 'someOtherName')
])

```

**Comment by stefanpenner** [comment]
actually, i think the following:

``js
function npmMain(packageName, outputFileName) {
  var resolvedPath = require.resolve(packageName);

  var inputPath = path.dirname(resolvedPath);
  var fileName = path.basename(resolvedPath);

  return new Funnel(inputPath, {
    files: [fileName],
    getDestinationPath: function() {
      return outputFileName || fileName;
    }
  });
}

```

**Comment by taras** [comment]
Let me update this

**Pull request closed without merging**